clients/include/testing_csrlsvqr.hpp

```diff
@@ -0,0 +1,639 @@
+        {
+            for(rocblas_int i = 0; i <= n; i++)
+                hptrA[0][i]++;
+
+            for(rocblas_int i = 0; i < nnzA; i++)
+                hindA[0][i]++;
+        }
+    }
+
+    if(GPU)
+    {
+        CHECK_HIP_ERROR(dptrA.transfer_from(hptrA));
+        CHECK_HIP_ERROR(dindA.transfer_from(hindA));
+        CHECK_HIP_ERROR(dvalA.transfer_from(hvalA));
+        CHECK_HIP_ERROR(dB.transfer_from(hB));
+    }
+}
+
+template <bool HOST, typename T, typename S, typename Td, typename Ud, typename Th, typename Uh>
+void csrlsvqr_getError(hipsolverSpHandle_t       handle,
+                       const int                 n,
+                       const int                 nnzA,
+                       const hipsparseMatDescr_t descrA,
+                       Ud&                       dptrA,
+                       Ud&                       dindA,
+                       Td&                       dvalA,
+                       Td&                       dB,
+                       const S                   tolerance,
+                       const int                 reorder,
+                       Td&                       dX,
+                       Uh&                       hptrA,
+                       Uh&                       hindA,
+                       Th&                       hvalA,
+                       Th&                       hB,
+                       Th&                       hX,
+                       Th&                       hXRes,
+                       Uh&                       hSingularity,
+                       double*                   max_err,
+                       const fs::path            testcase)
+{
+    // input data initialization
+    csrlsvqr_initData<true, true, T>(
+        handle, n, nnzA, descrA, dptrA, dindA, dvalA, dB, hptrA, hindA, hvalA, hB, hX, testcase);
+
+    // execute computations
+    // GPU lapack
+    CHECK_ROCBLAS_ERROR(hipsolver_csrlsvqr(HOST,
+                                           handle,
+                                           n,
+                                           nnzA,
+                                           descrA,
+                                           dvalA.data(),
+                                           dptrA.data(),
+                                           dindA.data(),
+                                           dB.data(),
+                                           tolerance,
+                                           reorder,
+                                           dX.data(),
+                                           hSingularity.data()));
+
+    CHECK_HIP_ERROR(hXRes.transfer_from(dX));
+
+    // compare computed results with original result
+    double err;
+    *max_err = 0;
+
+    err      = norm_error('I', n, 1, n, hX[0], hXRes[0]);
+    *max_err = err > *max_err ? err : *max_err;
+
+    // TODO: Add non-positive definite test matrices
+    // also check info for singularities
```

Comment:
Here what we need to add in the future is singular matrices.
The TODO comment needs to be updated.